File: Swift/125 Valid Palindrome.swift
class Solution {
    func isPalindrome(_ s: String) -> Bool {
        var str = s.filter { $0.isLetter || $0.isNumber }
        var res: [String] = []
        for e in str {
            if e.isUppercase {
                res.append(e.lowercased())
            } else {
                res.append(String(e))
            }
        }

        let n = res.count
        var i = 0, j = n - 1
        while i < j {
            if res[i] != res[j] {
                return false
            }
            i += 1
            j -= 1
        }
        return true
    }
}

File: C++/125 Valid Palindrome.cpp
using namespace std;

class Solution
{
public:
    bool isPalindrome(string s)
    {
        string res;
        for (auto &c : s)
        {
            if (isalpha(c) || isdigit(c))
            {
                if (isupper(c))
                    res += tolower(c);
                else
                    res += c;
            }
        }
        int n = res.size();
        int i = 0, j = n - 1;
        while (i < j)
        {
            if (res[i] != res[j])
                return false;
            i++;
            j--;
        }
        return true;
    }
};